src/pages/ZNS/ZNS.tsx

```diff
@@ -193,6 +194,11 @@ const ZNS: React.FC<ZNSProps> = ({ domain, version }) => {
 		if (triedEagerConnect)
 			addNotification(active ? 'Wallet connected.' : 'Wallet disconnected.');
 		// eslint-disable-next-line react-hooks/exhaustive-deps
+		
+	if(disconnectState === "true" && active!){
```

Comment:
Comparing the variable to the string "true" seems very dangerous from a maintainability perspective. Someone might see this later and accidently remove the "'s, or they might try to print `disconnectState` and then see that it's `false` and wonder why `!disconnectState === true`

Never expose a string that's the literal "true" "false" unless you want to confuse a programmer 3 months from now.